figures/views.py

```diff
@@ -392,6 +393,125 @@ def get_serializer_context(self):
         return context
 
 
+class CourseMonthlyMetricsViewSet(CommonAuthMixin, viewsets.ViewSet):
+    """
+
+    """
+    months_back = 6
+
+    def site_course_helper(self, pk):
+        """
+
+        Improvements:
+        * make this a decorator
+        * Test this with both course id strings and CourseKey objects
+        """
+        course_id = pk.replace(' ', '+')
```

Comment:
As far as I know we shouldn't be doing such replacement. It's the frontend responsibility to encode parameters so it passes `courses-v1+Appsembler+Demo+2020` as `courses-v1%2BAppsembler%2BDemo%2B2020` by encoding it with the JavaScript function as shown below:

```js
encodeURIComponent('courses-v1+Appsembler+Demo+2020')
```

I think it's fine to keep it for now, but recognize it's a bug to be resolved later.

Reply:
Thanks! @grozdanowski Please note this ^^^